waspc/src/Wasp/Generator/DbGenerator/Operations.hs

```diff
@@ -77,11 +77,17 @@ finalizeMigration genProjectRootDirAbs dbMigrationsDirInWaspProjectDirAbs onLast
 
 -- | Copies the DB migrations from the generated project dir back up to theh wasp project dir
 copyMigrationsBackToSource :: Path' Abs (Dir ProjectRootDir) -> Path' Abs (Dir DbMigrationsDir) -> IO (Either String ())
-copyMigrationsBackToSource genProjectRootDirAbs dbMigrationsDirInWaspProjectDirAbs =
-  copyDirectoryRecursive genProjectMigrationsDir waspMigrationsDir >> return (Right ())
-    `catch` (\e -> return $ Left $ show (e :: P.PathException))
-    `catch` (\e -> return $ Left $ show (e :: IOError))
+copyMigrationsBackToSource genProjectRootDirAbs dbMigrationsDirInWaspProjectDirAbs = do
+  doesMigrationsDirExist <- doesDirectoryExist $ SP.fromAbsDir genProjectMigrationsDir
+
+  if doesMigrationsDirExist
+    then copyMigrationsDir
+    else return $ Right ()
   where
+    copyMigrationsDir =
+      copyDirectoryRecursive genProjectMigrationsDir waspMigrationsDir >> return (Right ())
+        `catch` (\e -> return $ Left $ show (e :: P.PathException))
+        `catch` (\e -> return $ Left $ show (e :: IOError))
```

Comment:
```suggestion
copyMigrationsBackToSource genProjectRootDirAbs dbMigrationsDirInWaspProjectDirAbs = do
  doesDirectoryExist (SP.fromAbsDir genProjectMigrationsDir) >>= \case ->
    False -> return $ Right ()
    True -> copyDirectoryRecursive genProjectMigrationsDir waspMigrationsDir >> return (Right ())
        `catch` (\e -> return $ Left $ show (e :: P.PathException))
        `catch` (\e -> return $ Left $ show (e :: IOError))
```

Reply:
We could also consider renaming whole functoin to `copyMigrationsBackToSourceIfTheyExist`, but I am not sure, we can also skip it maybe.